### ISSUE TITLE
[Feature] Add Support for BTRFS Multi drive Devices (like RAID)

### ISSUE THREAD

**Issue by PlexSheep** [MauriceNino/dashdot#602]
### Description of the feature

BTRFS has a special Multi device functionality, like raid. My Homeserver has 3 drives working together to create something like a RAID1 array, all data is stored twice accross all devices, if one breaks all data can be restored.

In short it looks like this:
```
plex@homeserver /srv % doas btrfs filesystem show .
doas (plex@homeserver) password: 
Label: 'RAID1 ARRAY BTRFS main data of homeserver'  uuid: 59765ea5-e3b7-4ba2-b7b1-19acc65bc6ca
        Total devices 3 FS bytes used 1.05TiB
        devid    1 size 4.55TiB used 728.53GiB path /dev/sdc
        devid    2 size 4.55TiB used 727.53GiB path /dev/sdd
        devid    3 size 4.55TiB used 729.01GiB path /dev/sde

```

This multidevice can be mounted with any of it's part devices. I could mount it with `mount /dev/sdc .`, `mount /dev/sdd .` and so on.

In my dashdot interface, the disks are more or less treated like seperate drives (see appendix 1&2). The first one shows how much the entire array stores, but the max capacity is that of a single drive. The other two are shown as empty drives.

It would be nice, if dashdot would notice them being a array and only displaying a single drive with the right capacities.

### Additional context

Appendix 1:
Disks in split view: page 1

![Screenshot_20230225_143053](https://user-images.githubusercontent.com/58274330/221359596-836b469a-2b81-4435-885c-d62515332aaa.png)

Appendix 2:
Disks in split view: page 2

![Screenshot_20230225_143406](https://user-images.githubusercontent.com/58274330/221359751-2ff5c71a-5ea1-4a06-bfd9-041e910f544e.png)

Appendix 3:
my docker-compose.yml

```yml
version: '3.5'
services:
  dash:
    image: mauricenino/dashdot:latest
    container_name: dashdot
    restart: unless-stopped
    environment:
      - DASHDOT_ENABLE_CPU_TEMPS=true
      - DASHDOT_ALWAYS_SHOW_PERCENTAGES=true
      - DASHDOT_PAGE_TITLE=homeserver.box - Dashboard
      - DASHDOT_ENABLE_STORAGE_SPLIT_VIEW=true
      - DASHDOT_SHOW_HOST=true
      - DASHDOT_NETWORK_LABEL_LIST=type,speed_up,speed_down,interface_speed,public_ip
      - DASHDOT_OVERRIDE_OS=Debian GNU/Linux 11 (bullseye)
      - DASHDOT_OVERRIDE_ARCH=x86_64
    privileged: true
    ports:
      - 8001:3001
    volumes:
      - /:/mnt/host:ro
 ```

Notes:
The 3 last disks are identical, that's why they have the same name.

Dashdot runs in a docker container, the multidevice setup is done on the host. See Appendix 3.

**Comment by MauriceNino** [comment]
Hello there, thanks for reporting this issue. Could you please run the following commands and paste the output of the last one?

```
docker pull mauricenino/dashdot:dev
docker container run -it -d --name dashdot_test mauricenino/dashdot:dev
docker container exec -it dashdot_test yarn cli raw-data --storage
```

**Comment by PlexSheep** [comment]
Sure thing, here you go:


<details>
  <summary>Output:</summary>

```js
const disks =  [
  {
    device: '/dev/sda',
    type: 'SSD',
    name: 'OCZ-TRION150    ',
    vendor: 'ATA',
    size: 240057409536,
    bytesPerSector: null,
    totalCylinders: null,
    totalHeads: null,
    totalSectors: null,
    totalTracks: null,
    tracksPerCylinder: null,
    sectorsPerTrack: null,
    firmwareRevision: '12.2',
    serialNum: '',
    interfaceType: 'SATA',
    smartStatus: 'unknown',
    temperature: null
  },
  {
    device: '/dev/sdb',
    type: 'HD',
    name: 'WDC WD10EZRZ-00H',
    vendor: 'Western Digital',
    size: 1000204886016,
    bytesPerSector: null,
    totalCylinders: null,
    totalHeads: null,
    totalSectors: null,
    totalTracks: null,
    tracksPerCylinder: null,
    sectorsPerTrack: null,
    firmwareRevision: '1A01',
    serialNum: '',
    interfaceType: 'SATA',
    smartStatus: 'unknown',
    temperature: null
  },
  {
    device: '/dev/sdc',
    type: 'HD',
    name: 'ST5000LM000-2AN1',
    vendor: 'Seagate',
    size: 5000981078016,
    bytesPerSector: null,
    totalCylinders: null,
    totalHeads: null,
    totalSectors: null,
    totalTracks: null,
    tracksPerCylinder: null,
    sectorsPerTrack: null,
    firmwareRevision: '0001',
    serialNum: '',
    interfaceType: 'SATA',
    smartStatus: 'unknown',
    temperature: null
  },
  {
    device: '/dev/sdd',
    type: 'HD',
    name: 'ST5000LM000-2AN1',
    vendor: 'Seagate',
    size: 5000981078016,
    bytesPerSector: null,
    totalCylinders: null,
    totalHeads: null,
    totalSectors: null,
    totalTracks: null,
    tracksPerCylinder: null,
    sectorsPerTrack: null,
    firmwareRevision: '0001',
    serialNum: '',
    interfaceType: 'SATA',
    smartStatus: 'unknown',
    temperature: null
  },
  {
    device: '/dev/sde',
    type: 'HD',
    name: 'ST5000LM000-2AN1',
    vendor: 'Seagate',
    size: 5000981078016,
    bytesPerSector: null,
    totalCylinders: null,
    totalHeads: null,
    totalSectors: null,
    totalTracks: null,
    tracksPerCylinder: null,
    sectorsPerTrack: null,
    firmwareRevision: '0001',
    serialNum: '',
    interfaceType: 'SATA',
    smartStatus: 'unknown',
    temperature: null
  }
]
const sizes =  [
  {
    fs: 'overlay',
    type: 'overlay',
    size: 197193945088,
    used: 38508707840,
    available: 148593741824,
    use: 20.58,
    mount: '/',
    rw: false
  },
  {
    fs: '/dev/sda2',
    type: 'ext4',
    size: 24427446272,
    used: 5184233472,
    available: 17976430592,
    use: 22.38,
    mount: '/mnt/host',
    rw: false
  },
  {
    fs: '/dev/sda5',
    type: 'ext4',
    size: 1928785920,
    used: 94208,
    available: 1811935232,
    use: 0.01,
    mount: '/mnt/host/tmp',
    rw: true
  },
  {
    fs: '/dev/sda6',
    type: 'ext4',
    size: 197193945088,
    used: 38508707840,
    available: 148593741824,
    use: 20.58,
    mount: '/mnt/host/var',
    rw: true
  },
  {
    fs: '/dev/sda3',
    type: 'ext4',
    size: 9738321920,
    used: 973328384,
    available: 8248205312,
    use: 10.55,
    mount: '/mnt/host/home',
    rw: true
  },
  {
    fs: '/dev/sda1',
    type: 'vfat',
    size: 535801856,
    used: 6045696,
    available: 529756160,
    use: 1.13,
    mount: '/mnt/host/boot/efi',
    rw: true
  },
  {
    fs: '/dev/sdb1',
    type: 'ext4',
    size: 983348297728,
    used: 547351105536,
    available: 385970315264,
    use: 58.65,
    mount: '/mnt/host/mnt/sdb1',
    rw: true
  },
  {
    fs: '/dev/sdc',
    type: 'btrfs',
    size: 7501471617024,
    used: 1176167198720,
    available: 4218677219328,
    use: 21.8,
    mount: '/mnt/host/srv',
    rw: true
  }
]
const blocks =  [
  {
    name: 'sda',
    type: 'disk',
    fsType: '',
    mount: '',
    size: 240057409536,
    physical: 'SSD',
    uuid: '',
    label: '',
    model: 'OCZ-TRION150',
    serial: '',
    removable: false,
    protocol: 'sata',
    group: '',
    device: '/dev/sda'
  },
  {
    name: 'sdb',
    type: 'disk',
    fsType: '',
    mount: '',
    size: 1000204886016,
    physical: 'HDD',
    uuid: '',
    label: '',
    model: 'WDC WD10EZRZ-00H',
    serial: '',
    removable: false,
    protocol: 'sata',
    group: '',
    device: '/dev/sdb'
  },
  {
    name: 'sdc',
    type: 'disk',
    fsType: 'btrfs',
    mount: '/mnt/host/srv',
    size: 5000981078016,
    physical: 'HDD',
    uuid: '59765ea5-e3b7-4ba2-b7b1-19acc65bc6ca',
    label: 'RAID1 ARRAY BTRFS main data of homeserver',
    model: 'ST5000LM000-2AN1',
    serial: '',
    removable: false,
    protocol: 'sata',
    group: '',
    device: '/dev/sdc'
  },
  {
    name: 'sdd',
    type: 'disk',
    fsType: 'btrfs',
    mount: '',
    size: 5000981078016,
    physical: 'HDD',
    uuid: '59765ea5-e3b7-4ba2-b7b1-19acc65bc6ca',
    label: 'RAID1 ARRAY BTRFS main data of homeserver',
    model: 'ST5000LM000-2AN1',
    serial: '',
    removable: false,
    protocol: 'sata',
    group: '',
    device: '/dev/sdd'
  },
  {
    name: 'sde',
    type: 'disk',
    fsType: 'btrfs',
    mount: '',
    size: 5000981078016,
    physical: 'HDD',
    uuid: '59765ea5-e3b7-4ba2-b7b1-19acc65bc6ca',
    label: 'RAID1 ARRAY BTRFS main data of homeserver',
    model: 'ST5000LM000-2AN1',
    serial: '',
    removable: false,
    protocol: 'sata',
    group: '',
    device: '/dev/sde'
  },
  {
    name: 'sda1',
    type: 'part',
    fsType: 'vfat',
    mount: '/mnt/host/boot/efi',
    size: 536870912,
    physical: '',
    uuid: '677A-9717',
    label: '',
    model: '',
    serial: '',
    removable: false,
    protocol: '',
    group: '',
    device: '/dev/sda'
  },
  {
    name: 'sda2',
    type: 'part',
    fsType: 'ext4',
    mount: '/mnt/host',
    size: 25000148992,
    physical: '',
    uuid: '2f0b5f10-773b-4b89-8342-be7f0f90f46c',
    label: '',
    model: '',
    serial: '',
    removable: false,
    protocol: '',
    group: '',
    device: '/dev/sda'
  },
  {
    name: 'sda3',
    type: 'part',
    fsType: 'ext4',
    mount: '/mnt/host/home',
    size: 10000269312,
    physical: '',
    uuid: '70c9e7e0-b42c-4b0d-93b2-f8e426604719',
    label: '',
    model: '',
    serial: '',
    removable: false,
    protocol: '',
    group: '',
    device: '/dev/sda'
  },
  {
    name: 'sda4',
    type: 'part',
    fsType: 'swap',
    mount: '[SWAP]',
    size: 1024458752,
    physical: '',
    uuid: '4bba83c2-8ad5-4496-9f75-ce926da55be7',
    label: '',
    model: '',
    serial: '',
    removable: false,
    protocol: '',
    group: '',
    device: '/dev/sda'
  },
  {
    name: 'sda5',
    type: 'part',
    fsType: 'ext4',
    mount: '/mnt/host/tmp',
    size: 1999634432,
    physical: '',
    uuid: 'c97d7589-32c8-4cdb-9917-fd3c155e4029',
    label: '',
    model: '',
    serial: '',
    removable: false,
    protocol: '',
    group: '',
    device: '/dev/sda'
  },
  {
    name: 'sda6',
    type: 'part',
    fsType: 'ext4',
    mount: '/mnt/host/var',
    size: 201494364160,
    physical: '',
    uuid: 'd403143b-47fe-4818-a42c-3906632a06e7',
    label: '',
    model: '',
    serial: '',
    removable: false,
    protocol: '',
    group: '',
    device: '/dev/sda'
  },
  {
    name: 'sdb1',
    type: 'part',
    fsType: 'ext4',
    mount: '/mnt/host/mnt/sdb1',
    size: 1000202043392,
    physical: '',
    uuid: 'ea9b7736-2d12-48c7-ae3b-899271aea8b0',
    label: 'data-HDD',
    model: '',
    serial: '',
    removable: false,
    protocol: '',
    group: '',
    device: '/dev/sdb'
  }
]
```

</details>

**Comment by MauriceNino** [comment]
Unfortunately, there is no data in there, which would allow me to group those drives. I would suggest using [DASHDOT_FS_DEVICE_FILTER](https://getdashdot.com/docs/config/features#dashdot_fs_device_filter) to filter out the duplicated empty devices. Would that work for you?
